packages/chrono-mongo-datastore/src/chrono-mongo-datastore.ts

```diff
@@ -114,6 +114,31 @@ export class ChronoMongoDatastore<TaskMapping extends TaskMappingBase>
     throw new Error(`Failed to insert ${String(input.kind)} document`);
   }
 
+  async delete<TaskKind extends keyof TaskMapping>(
+    taskId: string,
+  ): Promise<Task<TaskKind, TaskMapping[TaskKind]> | undefined> {
+    const task = await this.database
+      .collection<TaskDocument<TaskKind, TaskMapping[TaskKind]>>(this.config.collectionName)
+      .findOneAndDelete({
+        _id: new ObjectId(taskId),
+        status: TaskStatus.PENDING,
+      });
+
+    if (!task) {
+      const existingTask = await this.database
+        .collection<TaskDocument<TaskKind, TaskMapping[TaskKind]>>(this.config.collectionName)
+        .findOne({
+          _id: new ObjectId(taskId),
+        });
+
+      if (existingTask) {
+        throw new Error(`Task ${taskId} has a ${existingTask.status} status and can not be deleted.`);
+      }
+    }
```

Comment:
This query may have concurrent issue where as some process already updated the status to `PENDING` (eg. due to task retry). Maybe we can just throw an error like below:
```suggestion
    if (!task) {
      throw new Error(`Task ${taskId} can not be deleted as it may not exist or it's not in PENDING status.`);
    }
```

Reply:
Never mind, mis remembered the code, your solution is consistent. I can make the in-memory store also throw on attempting to delete a non-existing task if that's preferrable.

Reply:
Yup, I think it also simplifies the implementation. We can revisit the non-existing task in the future if there is a use case that we need granular error handling for it.

Reply:
Makes sense, thanks!